File: csharp/crypto-square/CryptoSquare.cs
using System;
using System.Linq;
using System.Collections.Generic;
using static System.Math;

public static class CryptoSquare
{
    public static string Ciphertext(string plaintext)
    {
        var clean = new string(plaintext.ToLower().Where(char.IsLetterOrDigit).ToArray());
        if(clean.Length == 0)
            return "";

        var sqrt = Sqrt((float)clean.Length);
        var (cols, rows) = ((int)Floor(sqrt), (int)Ceiling(sqrt));
        if(cols * rows < clean.Length) 
            cols++;

        var square = Enumerable.Range(0, cols).Select(i => 
            new string(clean.Skip(i * rows).Take(rows).ToArray()).PadRight(rows))
                .ToArray();

        var cipher = Enumerable.Range(0, rows).Select(c => 
            new string(Enumerable.Range(0, cols)
                .Select(r => square[r][c]).ToArray()));

        return string.Join(' ', cipher);
    }
}

File: go/crypto-square/crypto_square.go
package cryptosquare

import (
	"math"
	"strings"
	"unicode"
)

// Encode produces a cypher using the crypto square algorithm
func Encode(raw string) string {
	var cleaner strings.Builder
	for _, char := range raw {
		if unicode.IsNumber(char) || unicode.IsLetter(char) {
			cleaner.WriteRune(unicode.ToLower(char))
		}
	}
	clean := []rune(cleaner.String())
	length := len(clean)

	root := math.Sqrt(float64(length))
	cols := int(math.Ceil(root))
	rows := cols - 1
	if rows*cols < length {
		rows++
	}

	var cypher strings.Builder
	for c := 0; c < cols; c++ {
		for r := 0; r < rows; r++ {
			index := r*cols + c
			if index >= length {
				cypher.WriteRune(' ')
			} else {
				cypher.WriteRune(clean[index])
			}
		}
		if c != cols-1 {
			cypher.WriteRune(' ')
		}
	}

	return cypher.String()
}
